### DISCUSSION TITLE
Android support

### DISCUSSION THREAD

**Discussion by TehDomic** [LizardByte/community#99] [Sunshine]
Is there any technical limitations or simply not anyone who's had the time to look at it yet? Lots of ARM64 devices out there, some even computational powerhouses.

**Comment by ReenigneArcher** [comment]
We have arm64/aarch64 support already.

**Reply by TehDomic** [reply]
Hm 🤔 yes, but only as a flatpak.
I'm trying to get it running on Android, (Termux) and flatpak apparently requires 'systemd'.
Are there any regular arm64 Linux builds available somewhere? .deb (or .tar.gz even) ?

**Reply by TehDomic** [reply]
Heck, even just straight up Android builds/forks would be an excellent idea. (in theory)

**Reply by ReenigneArcher** [reply]
No... you're in uncharted territory trying to run Sunshine on Android.

**Reply by TehDomic** [reply]
Oh okay. Just thought since Vulkan (Zink), GL4ES, and other projects are doing so well you've already tested everything available and maybe it ran sunshine well.

**Reply by yonggamer** [reply]
is it going to happen or not?

**Reply by ReenigneArcher** [reply]
Very unlikely

**Reply by yonggamer** [reply]
why is it Very unlikely?

**Reply by ReenigneArcher** [reply]
No one is working on it, and it's unlikely to perform well even if someone did work on it.

**Reply by yonggamer** [reply]
why is it unlikely to perform well even if someone did work on it ?
[this project](https://github.com/Genymobile/scrcpy) has good performance with screen sharing

**Reply by ReenigneArcher** [reply]
Encoding is slow without a dedicated GPU and high performance encoder. This is evident by the fact they are 2 frames behind the host at 1080p. https://github.com/Genymobile/scrcpy/pull/646

Not sure what device they're using, but it's likely to be worse with lower end devices.

Still, looks like a cool project. Why not use that? Instead of trying to force Sunshine to be something that it's not?

**Reply by ReenigneArcher** [reply]
Correct me if I'm wrong, but as far as I understand how `scrcpy` works... Nothing is actually installed to the Android device. The client device just connects directly to the device using ADB. In theory this functionality could be added to Sunshine, but it would add an extra layer of latency. Basically Sunshine would act as a proxy when connecting to an Android device.

However; going this route seems kind of pointless IMO. It would be far easier to use `scrcpy` on your Sunshine host, and then have Sunshine stream that. Since Sunshine allows for custom start and prep commands, it would be pretty easy to integrate as a game/app in Sunshine.

**Comment by Laurie-Lin** [comment]
@ReenigneArcher hi, I started by perfecting my sunshine_android, there was still a lot to do, and I also joined discord

**Comment by ii0** [comment]
genymobile/scrcpy can only  Android to PC,  it cannot  Android to Android.
Could anyone tell me the best solution for Android to Android ? Thank you very much!

**Comment by ReenigneArcher** [comment]
There's a long way to go, but some initial work has begun in https://github.com/LizardByte/Sunshine/pull/3741
